Cython/Utility/ObjectHandling.c

```diff
@@ -2931,7 +2943,22 @@ static CYTHON_INLINE int __Pyx_object_dict_version_matches(PyObject* obj, PY_UIN
 
 /////////////// PyMethodNew.proto ///////////////
 
-#if PY_MAJOR_VERSION >= 3
+#if CYTHON_COMPILING_IN_LIMITED_API
+static PyObject *__Pyx_PyMethod_New(PyObject *func, PyObject *self, PyObject *typ) {
```

Comment:
This also looks really slow, given that we'd do this on each method lookup. It's ok to get it working this way, but I doubt that it can stay that way.

In general, we should try not to be visibly slower than CPython, at least on stuff that happens repeatedly. We can assume that users of the Limited API do not care as much about performance as others, but they'd probably still not want their code to run much slower than in plain Python.

Reply:
It might be worth double checking if we actually use this. I think it's mainly used in CythonFunction.c, which has been rewritten substantially for the limited API anyway, so it's possible that we can just drop this.